torch/autograd/profiler.py

```diff
@@ -606,9 +605,7 @@ def __init__(self, name: str):
         self.name: str = name
         # Whether or not we should run record function's end callbacks when exiting.
         self.run_callbacks_on_exit: bool = True
-        # Stores underlying RecordFunction as a tensor. TODO: move to custom
-        # class (https://github.com/pytorch/pytorch/issues/35026).
-        self.handle: torch.Tensor = torch.zeros(1)
+        self.handle : __torch__.torch.classes.profiler._RecordFunctionHolder = torch.ops.profiler._record_function_enter('placeholder')
```

Comment:
We should only call `_record_function_enter` in the `__enter__` part of the context manager, not during init time right? Here we should probably just initialize `self.handle: __torch__.classes.profiler.RecordFunctionHolder = RecordFunctionHolder()`? This may also fix `test_profiler_propagation`.

Reply:
Unfortunately seems if we use CustomClassObject.member_function, torch.jit.trace will throw an exception (the same as before), pls see https://docs.google.com/document/d/1_ZcjIyIHKRbe1_dXYh7oTqTn0APEfnEEu1AySQwd3Qk/edit#heading=h.si9n17vbfpv1 for context

Reply:
Anyway changed back to member function to unhide the potential JIT issue.